input-output-hk/ouroboros-network
# Fix and unify stylish-haskell checks

Previously, the nix stylish-haskell checks where only checking that the files *can* be formatted (no errors due to e.g. CPP), but not that they actually are formatted, as `stylish-haskell -i` will succeed with exit code 0 even if it some of its input files were not already formatted. This PR fixes this by using the `exit_code: error_on_format` of stylish-haskell.

In addition:
  - We reuse the scripts in `scripts/ci` such that people can easily run stylish-haskell in the same way as it is checked in CI.
  - We use `cleanGit` to only copy the git-tracked files to the nix store. Previously, the entire local development directory would be copied (easily >1GB). This subsumes #3669.
  - `checks.styles.check-stylish-network` is removed and part of `nix-build -A checks.styles.check-stylish`.

To test this:
  1. Introduce some change which makes the file unformatted.
  2. Check that `nix-build -A checks.styles.check-stylish` fails.
